test/new-e2e/utils/e2e/client/ssh_client.go

```diff
@@ -7,29 +7,40 @@ package client
 
 import (
 	"fmt"
+	"testing"
 	"time"
 
 	"github.com/DataDog/datadog-agent/test/new-e2e/utils/clients"
 	"github.com/DataDog/test-infra-definitions/common/utils"
+	"github.com/stretchr/testify/require"
 	"golang.org/x/crypto/ssh"
 )
 
 type sshClient struct {
 	client *ssh.Client
+	t      *testing.T
 }
 
-func newSSHClient(sshKey string, connection *utils.Connection) (*sshClient, error) {
+func newSSHClient(t *testing.T, sshKey string, connection *utils.Connection) (*sshClient, error) {
 	client, _, err := clients.GetSSHClient(
 		connection.User,
 		fmt.Sprintf("%s:%d", connection.Host, 22),
 		sshKey,
 		2*time.Second, 5)
 	return &sshClient{
 		client: client,
+		t:      t,
 	}, err
 }
 
-// Execute a command
-func (vm *sshClient) Execute(command string) (string, error) {
-	return clients.ExecuteCommand(vm.client, command)
+// ExecuteWithError executes a command and returns an error if any.
+func (sshClient *sshClient) ExecuteWithError(command string) (string, error) {
+	return clients.ExecuteCommand(sshClient.client, command)
+}
+
+// Execute execute a command and asserts there is no error.
+func (sshClient *sshClient) Execute(command string) string {
```

Comment:
<!-- Thoughts represent an idea that popped up from reviewing. These comments are non-blocking by nature, but they are extremely valuable and can lead to more focused initiatives and mentoring opportunities. -->
**💭 thought**
Opinionated, I would rather call this one `ExecuteNoError` or `ExecuteRequireNoError` and leave the other one alone. I would also have `t` passed to `Execute` rather than keeping it in the `sshClient` struct, even if I can see that being part of testing, it is clear it is used in tests.

Reply:
As discussed offline, the struct is not well named as this is not a generic ssh client but a way to connect to the VM. I will rename it.